`panic: runtime error: slice bounds out of range [:1] with capacity 0` with input `0d1d01`

Here's a minimal reproduction of the issue:

```go
package main

import (
	"fmt"

	"github.com/justinian/dice"
)

func main() {
	_, output, err := dice.Roll("0d1d01")
	fmt.Println("Output:", output)
	fmt.Println("Error:", err)
}
```

Playground link: https://go.dev/play/p/nGJLTC3dV_T

And the full output:
```
panic: runtime error: slice bounds out of range [:1] with capacity 0

goroutine 1 [running]:
github.com/justinian/dice.StdRoller.Roll({}, {0xc000116400, 0x8, 0xc00006a180?})
	/tmp/gopath3705264433/pkg/mod/github.com/justinian/dice@v1.0.2/std.go:90 +0x515
github.com/justinian/dice.Roll({0x4e1269, 0x6})
	/tmp/gopath3705264433/pkg/mod/github.com/justinian/dice@v1.0.2/dice.go:74 +0xd1
main.main()
	/tmp/sandbox2393286976/prog.go:10 +0x25

Program exited.
```

I'm aware that the input is an invalid format, but I would expect an error message rather than a panic.

--- a/dice_test.go
+++ b/dice_test.go
@@ -51,6 +51,39 @@ func TestRoll(t *testing.T) {
 	if _, ok := res.(VsResult); !ok {
 		t.Fatalf("%s is not a VsResult", roll)
 	}
+
+	// Trying to keep or drop too many dice
+	roll = "2d6k5"
+	_, _, err = Roll(roll)
+	if err != nil {
+		t.Logf("err '%v' properly detected in %s", err, roll)
+	} else {
+		t.Fatalf("err not detected in %s", roll)
+	}
+
+	roll = "2d6kl5"
+	_, _, err = Roll(roll)
+	if err != nil {
+		t.Logf("err '%v' properly detected in %s", err, roll)
+	} else {
+		t.Fatalf("err not detected in %s", roll)
+	}
+
+	roll = "2d6dh5"
+	_, _, err = Roll(roll)
+	if err != nil {
+		t.Logf("err '%v' properly detected in %s", err, roll)
+	} else {
+		t.Fatalf("err not detected in %s", roll)
+	}
+
+	roll = "2d6dl5"
+	_, _, err = Roll(roll)
+	if err != nil {
+		t.Logf("err '%v' properly detected in %s", err, roll)
+	} else {
+		t.Fatalf("err not detected in %s", roll)
+	}
 }
 
 func TestText(t *testing.T) {

--- a/std.go
+++ b/std.go
@@ -82,19 +82,33 @@ func (StdRoller) Roll(matches []string) (RollResult, error) {
 	case "k":
 		fallthrough
 	case "kh":
-		result.Dropped = result.Rolls[:size-num]
-		result.Rolls = result.Rolls[size-num:]
+		slice := size - num
+		if slice < 0 {
+			return nil, errors.New("Can't keep more dice than rolled")
+		}
+		result.Dropped = result.Rolls[:slice]
+		result.Rolls = result.Rolls[slice:]
 	case "d":
 		fallthrough
 	case "dl":
+		if num > size {
+			return nil, errors.New("Can't drop more dice than rolled")
+		}
 		result.Dropped = result.Rolls[:num]
 		result.Rolls = result.Rolls[num:]
 	case "kl":
+		if num > size {
+			return nil, errors.New("Can't keep more dice than rolled")
+		}
 		result.Dropped = result.Rolls[num:]
 		result.Rolls = result.Rolls[:num]
 	case "dh":
-		result.Dropped = result.Rolls[size-num:]
-		result.Rolls = result.Rolls[:size-num]
+		slice := size - num
+		if slice < 0 {
+			return nil, errors.New("Can't drop more dice than rolled")
+		}
+		result.Dropped = result.Rolls[slice:]
+		result.Rolls = result.Rolls[:slice]
 	}
 
 	for i := 0; i < len(result.Rolls); i++ {

